### PR TITLE
Comment about deprecating code in Rails

### PR DESCRIPTION
**Pull request by kdsingharneja**
@AaronV

### CODE REVIEW

File: style/README.md [hunk 1]
```diff
@@ -346,3 +346,12 @@ Testing
   appear in the class.
 
 [Imperative mood]: http://en.wikipedia.org/wiki/Imperative_mood
+
+Legacy Code
```
Comment (jeremyw) [inline style/README.md]: Is there a spot for this other than a new "legacy code" section (deprecated != legacy)?

File: style/README.md [hunk 2]
```diff
@@ -346,3 +346,12 @@ Testing
...
+-----------
+### Rails
+
+* Use the [deprecate module] in Rails to deprecate code. Make
```
Comment (jeremyw) [inline style/README.md]: Extra "Make"

File: style/README.md [hunk 3]
```diff
@@ -346,3 +346,12 @@ Testing
...
+* Also a readable comment in the code, if you are not going to generate a warning
```
Comment (jeremyw) [inline style/README.md]: I'm not sure what this means.

File: style/README.md (hunk 1 above)
Comment (kdsingharneja) [inline style/README.md]: Should be clearer now.

I did not see one, hence added this here.

Comment (thegreatape) [inline style/README.md]: I think 99% of the time, we shouldn't be deprecating methods at all. We've already got too much of a bad habit of leaving `TODO: change this` comments in the code around for a long time, and I don't see marking methods as deprecated any differently. Let's just make the change if something needs to change.

The only exception would be API endpoint used outside the company (or perhaps used by the mobile app), but those won't be helped by the Rails `deprecate` method.

Comment (jeremyw) [inline style/README.md]: I agree with @thegreatape. Is there a case where we should be using `deprecate` now or in the past?

**Comment by AaronV** [comment]
I think what @thegreatape said sounds right. Better to just remove stuff we don't need, instead adding comments and hooks nobody will ever clean up. Let's try to avoid adding a gem to handle this for now.

**Comment by kdsingharneja** [comment]
I am cool with that.

@AaronV Lets remove the code you were going to deprecate if it does not need to be there?!

**Comment by kdsingharneja** [comment]
closing PR...appears we have it covered!

**Pull request closed without merging**